hudi-common/src/main/java/org/apache/hudi/metadata/HoodieBackedTableMetadata.java

```diff
@@ -305,6 +305,10 @@ private Map<String, HoodieRecord<HoodieMetadataPayload>> lookupKeysFromFileSlice
       return readFromBaseAndMergeWithLogRecords(baseFileReader, sortedKeys, fullKeys, logRecords, timings, partitionName);
     } catch (IOException ioe) {
       throw new HoodieIOException("Error merging records from metadata table for  " + keys.size() + " key : ", ioe);
+    } finally {
```

Comment:
should we move this to getRecordsByKeys and getRecordsByKeyPrefixes since thats where we call
```
partitionFileSliceMap.computeIfAbsent(partitionName,
        k -> HoodieTableMetadataUtil.getPartitionLatestMergedFileSlices(metadataMetaClient, metadataFileSystemView, partitionName));
```

Reply:
The current code is better, because the readers are initialized in `lookupKeysFromFileSlice`.

Reply:
yeah, as @danny0405 said, where to initialize and where to close